read-fonts/src/tables/hvar.rs

```diff
@@ -0,0 +1,98 @@
+//! The [HVAR (Horizontal Metrics Variation)](https://docs.microsoft.com/en-us/typography/opentype/spec/hvar) table
+
+use crate::variations::{DeltaSetIndex, DeltaSetIndexMap, ItemVariationStore};
+use font_types::Tag;
+
+/// 'HVAR'
+pub const TAG: Tag = Tag::new(b"HVAR");
+
+include!("../../generated/generated_hvar.rs");
+
+impl<'a> Hvar<'a> {
+    /// Returns the advance width delta for the specified glyph identifier and
+    /// normalized variation coordinates.
+    pub fn advance_width_delta(
+        &self,
+        glyph_id: GlyphId,
+        coords: &[F2Dot14],
+    ) -> Result<Fixed, ReadError> {
+        let gid = glyph_id.to_u16();
+        let ix = match self.advance_width_mapping() {
+            Some(Ok(dsim)) => dsim.get(gid as u32)?,
+            _ => DeltaSetIndex {
+                outer: 0,
+                inner: gid,
+            },
+        };
+        let ivs = self.item_variation_store()?;
+        ivs.compute_delta(ix, coords)
+    }
+
+    /// Returns the left side bearing delta for the specified glyph identifier and
+    /// normalized variation coordinates.
+    pub fn lsb_delta(&self, glyph_id: GlyphId, coords: &[F2Dot14]) -> Result<Fixed, ReadError> {
+        let gid = glyph_id.to_u16();
+        let ix = match self.lsb_mapping() {
+            Some(Ok(dsim)) => dsim.get(gid as u32)?,
+            _ => return Err(ReadError::NullOffset),
+        };
+        let ivs = self.item_variation_store()?;
+        ivs.compute_delta(ix, coords)
+    }
+
+    /// Returns the left side bearing delta for the specified glyph identifier and
+    /// normalized variation coordinates.
+    pub fn rsb_delta(&self, glyph_id: GlyphId, coords: &[F2Dot14]) -> Result<Fixed, ReadError> {
+        let gid = glyph_id.to_u16();
+        let ix = match self.rsb_mapping() {
+            Some(Ok(dsim)) => dsim.get(gid as u32)?,
+            _ => return Err(ReadError::NullOffset),
+        };
+        let ivs = self.item_variation_store()?;
+        ivs.compute_delta(ix, coords)
+    }
+}
```

Comment:
Hvar lsb_delta, rsb_delta and Vvar txb_delta, bsb_delta appear to be almost identical and in need of a common helper.

They are also very close to advance_(width|height)_delta, differing only in what happens if the mapping is None. Looks like a common function whose parameters include a fn to handle the mapping None case might go quite far?

Reply:
Will do!

Reply:
Done